oops/oops.go

```diff
@@ -469,3 +470,61 @@ func Recover(p interface{}) error {
 	}
 	return wrapf(fmt.Errorf("recovered panic: %v", p), "")
 }
+
+// MultiError acts as a container to capture multiple errors and their tracebacks
+// When this object is written to a log it will log all the errors it has been fed.
+type multiError struct {
+	errors []error
+}
+
+func (omr *multiError) Error() string {
+	var b strings.Builder
+	b.WriteString("the following errors occurred:\n\n")
+	for _, e := range omr.errors {
+		if oerr, ok := e.(*oopsError); ok {
+			oerr.writeStackTrace(&b)
+		} else {
+			b.WriteString(e.Error())
+		}
+		b.WriteString("\n\n---\n\n")
+	}
+
+	return b.String()
+}
+
+// Return the slice of errors contained within a MultiError instance, or a slice containing
+// the passed error if the argument is a single error
+func Errors(err error) []error {
+	if omr, ok := err.(*multiError); ok {
+		return omr.errors
+	}
+	return []error{err}
+}
+
+func (omr *multiError) Errors() []error {
+	return omr.errors
+}
+
+// Concatentate an error with another, yielding a MultiError instance
```

Comment:
Nit:

```suggestion
// Append concatenates an error onto another, yielding a multiError instance
```

...although, I think it's worth ironing out whether `multiError` ought to be publicly exported and exposed as part of `oops`'s API as `MultiError`, or whether its implementation should remain private, unexposed, and therefore unmentioned in public API code comments/documentation. Do you have thoughts on that, @bharling ?

Reply:
I did have it publicly exposed before but opted to follow the pattern for oopsError and uber's multiError, both of which only export the functions needed to create the underlying structs ( `Wrapf`, `Errorf`, `Append`, `Combine` etc. ). Since the oops.multiError implements `Errors()` IMO the actual struct could probably remain private?